### PR TITLE
UnaryOps: Don't use buffer for certain sizes

### PR DESCRIPTION
**Pull request by cpuhrsch**
Timings follow soon.

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -261,7 +261,13 @@ def test_linear_algebra_scalar_raises(self):
         self.assertRaises(RuntimeError, lambda: torch.addr(m, v, s))
         self.assertRaises(RuntimeError, lambda: torch.addr(m, s, v))
 
-    def _test_math(self, torchfn, mathfn, input=None):
+    def _test_math(self, torchfn, mathfn, input=None, nonzero=False, offset=0):
+        def proc(input):
```
Comment (ezyang) [inline test/test_torch.py]: What does the name proc mean?

Comment (cpuhrsch) [reply test/test_torch.py]: "process" or "preprocess". I can choose a longer name.

**Comment by ezyang** [comment]
@cpuhrsch I'm going to close this for now so it gets out of the "external contrib" queue. Please feel free to reopen when relevant.

**Pull request closed without merging**